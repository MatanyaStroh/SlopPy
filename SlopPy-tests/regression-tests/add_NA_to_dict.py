# try to add an NA object to a dict in various ways;
# all attempts should get rejected

x = 1 / 0
assert type(x) is NA

# dict constructor should ignore all instances of x in key or value
assert {x:1} == {}
assert {1:x} == {}
assert {x:1, 'hello' : 'world'} == {'hello' : 'world'}
assert {1:x, 'hello' : 'world'} == {'hello' : 'world'}

d = {}
d['bob'] = x
assert len(d) == 0
d[x] = 'joe'
assert len(d) == 0
d.update({x:1})
assert len(d) == 0
y = d.setdefault('unknown', x)
assert y is x
assert len(d) == 0

