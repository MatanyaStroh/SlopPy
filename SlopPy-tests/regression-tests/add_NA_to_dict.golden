{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/add_NA_to_dict.py", line 4, in <module>
    x = 1 / 0
}
{STORE_MAP(key=NA) | regression-tests/add_NA_to_dict.py <module>:8}
{STORE_MAP(value=NA) | regression-tests/add_NA_to_dict.py <module>:9}
{STORE_MAP(key=NA) | regression-tests/add_NA_to_dict.py <module>:10}
{STORE_MAP(value=NA) | regression-tests/add_NA_to_dict.py <module>:11}
{PyObject_SetItem(value=NA) | regression-tests/add_NA_to_dict.py <module>:14}
{PyObject_SetItem(key=NA) | regression-tests/add_NA_to_dict.py <module>:16}
{STORE_MAP(key=NA) | regression-tests/add_NA_to_dict.py <module>:18}
{dict.setdefault(*, NA) | regression-tests/add_NA_to_dict.py <module>:20}
{dict_init(value=NA) | regression-tests/add_NA_to_dict.py <module>:24}
{dict_init(value=NA) | regression-tests/add_NA_to_dict.py <module>:25}
{dict_init(value=NA) | regression-tests/add_NA_to_dict.py <module>:25}
{dict_init(value=NA) | regression-tests/add_NA_to_dict.py <module>:26}
{dict_init(value=NA) | regression-tests/add_NA_to_dict.py <module>:26}
{dict_init(value=NA) | regression-tests/add_NA_to_dict.py <module>:27}
{dict_init(value=NA) | regression-tests/add_NA_to_dict.py <module>:27}
